OpenEdXMobile/src/main/java/org/edx/mobile/util/VideoUtil.java

```diff
@@ -0,0 +1,47 @@
+package org.edx.mobile.util;
+
+import android.support.annotation.NonNull;
+
+import org.edx.mobile.model.course.VideoData;
+import org.edx.mobile.model.course.VideoInfo;
+
+public class VideoUtil {
+    /**
+     * Check the validity of video url.
+     *
+     * @param videoUrl Url which needs to be validated.
+     * @return <code>true</code> if video url is valid.
+     */
+    public static boolean isValidVideoUrl(@NonNull String videoUrl) {
+        if (videoUrl.contains("edx-course-videos")) {
+            return true;
+        }
+        for (final String format : AppConstants.SUPPORTED_VIDEO_FORMATS) {
+            if (isVideoHasFormat(videoUrl, format)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean isM3U8FormatVideo(@NonNull String videoUrl) {
+        return isVideoHasFormat(videoUrl, AppConstants.VIDEO_FORMAT_M3U8);
+    }
+
+    public static boolean isVideoHasFormat(@NonNull String videoUrl, @NonNull String videoFormat) {
+        return videoUrl.endsWith(videoFormat);
+    }
+
+    /**
+     * Validate if video is considerable to download.
+     *
+     * @param video Data of the video which needs to be validated for downloadable.
```

Comment:
`downloadability` would be the right word here instead of `downloadable`.

Reply:
downloadability is not a right english word, changed to
`Data of the video whose download validity needs to be checked.`